pkg/controller/sriovoperatorconfig/sriovoperatorconfig_controller.go

```diff
@@ -0,0 +1,282 @@
+	return nil
+}
+
+func (r *ReconcileSriovOperatorConfig) deleteWebhookObject(obj *uns.Unstructured) error {
+	if err := r.deleteK8sResource(obj); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *ReconcileSriovOperatorConfig) syncWebhookObject(dc *sriovnetworkv1.SriovOperatorConfig, obj *uns.Unstructured) error {
+	var err error
+	logger := log.WithName("syncWebhookObject")
+	logger.Info("Start to sync Objects")
+	scheme := kscheme.Scheme
+	switch kind := obj.GetKind(); kind {
+	case "MutatingWebhookConfiguration":
+		whs := &admissionregistrationv1beta1.MutatingWebhookConfiguration{}
+		err = scheme.Convert(obj, whs, nil)
+		r.syncWebhook(dc, whs)
+		if err != nil {
+			logger.Error(err, "Fail to sync mutate webhook")
+			return err
+		}
+	case "ConfigMap":
+		cm := &corev1.ConfigMap{}
+		err = scheme.Convert(obj, cm, nil)
+		err = r.syncWebhookConfigMap(dc, cm)
+		if err != nil {
+			logger.Error(err, "Fail to sync webhook config map")
+			return err
+		}
+	case "ServiceAccount", "DaemonSet", "Service", "ClusterRole", "ClusterRoleBinding":
+		err = r.syncK8sResource(dc, obj)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileSriovOperatorConfig) syncWebhook(cr *sriovnetworkv1.SriovOperatorConfig, in *admissionregistrationv1beta1.MutatingWebhookConfiguration) error {
+	logger := log.WithName("syncWebhook")
+	logger.Info("Start to sync webhook", "Name", in.Name, "Namespace", in.Namespace)
+
+	if err := controllerutil.SetControllerReference(cr, in, r.scheme); err != nil {
+		return err
+	}
+	whs := &admissionregistrationv1beta1.MutatingWebhookConfiguration{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: in.Name}, whs)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.client.Create(context.TODO(), in)
+			if err != nil {
+				return fmt.Errorf("Couldn't create webhook: %v", err)
+			}
+			logger.Info("Create webhook for", in.Namespace, in.Name)
+		} else {
+			return fmt.Errorf("Fail to get webhook: %v", err)
+		}
+	}
+
+	// Note:
+	// we don't need to manage the update of MutatingWebhookConfiguration here
+	// as it's handled by caconfig controller
+
+	return nil
+}
+
+func (r *ReconcileSriovOperatorConfig) syncWebhookConfigMap(cr *sriovnetworkv1.SriovOperatorConfig, in *corev1.ConfigMap) error {
+	logger := log.WithName("syncWebhookConfigMap")
+	logger.Info("Start to sync config map", "Name", in.Name, "Namespace", in.Namespace)
+
+	if err := controllerutil.SetControllerReference(cr, in, r.scheme); err != nil {
+		return err
+	}
+	cm := &corev1.ConfigMap{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: in.Namespace, Name: in.Name}, cm)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.client.Create(context.TODO(), in)
+			if err != nil {
+				return fmt.Errorf("Couldn't create config map: %v", err)
+			}
+			logger.Info("Create config map for", in.Namespace, in.Name)
+		} else {
+			return fmt.Errorf("Fail to get config map: %v", err)
+		}
+	}
+
```

Comment:
I think we shall make sure the annotation of the CM is intact. Update it to the correct value when needed.

Reply:
do you mean update the CM here? (I don't know if there is a way to just update annotation)

Reply:
You can just reuse the object `cm` from `Get` call, and modify the annotation when need.

Reply:
fixed